### ISSUE TITLE
tpm2_getekcertificate: add support for AMD TPM

### ISSUE THREAD

**Issue by XeR** [tpm2-software/tpm2-tools#3158]
Hello,

`tpm2_getekcertificate` is a tool to "Retrieve the Endorsement key Certificate".
If the target TPM is a firmware TPM, it will fetch the certificate from Intel's servers.
This does not work if the TPM was not manufactured by Intel.

`tpm2_getekcertificate` should be able to retrieve EK certificates from other manufacturers.

Here is how to obtain the certificates for a TPM with a `TPM2_PT_MANUFACTURER` of 0x414D4400 (AMD) :

For RSA:
```sh
# Create an RSA EK
tpm2_createek -c /dev/null -f pem -u rsa.pem -G rsa

# Get the modulus (have faith)
eval $(openssl rsa -pubin -noout -modulus < rsa.pem)

# hash = sha256(00 00 22 22 || (uint32_t)exp || modulus)[0..16] (in hex)
hash=$(echo 00002222 00010001 $Modulus | xxd -r -ps | openssl dgst -sha256 -binary | head -c 16 | xxd -ps)

# Retrieve certificate
curl -O "https://ftpm.amd.com/pki/aia/$hash"
```

For ECC:
```sh
# Create an ECC EK
tpm2_createek -c /dev/null -f pem -u ecc.pem -G ecc

# Get the public key blob
pub=$(openssl ec -pubin -text -noout < ecc.pem | grep '^ ')

# Convert to binary, skip initial 0x04 (https://stackoverflow.com/a/44827409)
bin=$(echo $pub | xxd -r -ps | tail -c +2)

# hash = sha256(00 00 44 44 || public key)
hash=$(printf '\x00\x00\x44\x44%s' "$bin" | openssl dgst -sha256 -binary | head -c 16 | xxd -ps)

# Retrieve certificate
curl -O "https://ftpm.amd.com/pki/aia/$hash"
```

Tested on a Zen 2 chip

**Comment by williamcroberts** [comment]
The newer EK tool is a bit more generic, but we'd also take a patch since you have the hardware to test it against.

**Comment by idesai** [comment]
@XeR you can specify where to look, "https://ftpm.amd.com/pki/aia/",  for the certificate as an argument.

**Comment by williamcroberts** [comment]
> @XeR you can specify where to look, "https://ftpm.amd.com/pki/aia/", for the certificate as an argument.

@idesai from their example it looks like a GET request to that URL with a hash at the end:
```bash
curl -O "https://ftpm.amd.com/pki/aia/$hash"
```

Did I miss something?

**Comment by williamcroberts** [comment]
> ```shell
> https://stackoverflow.com/a/44827409
> ```

I think the fix is in get_tpm_properties add a check for the vendor being AMD, set a flag in the tool context structure and in `get_web_ek_certificate` we use the check to change the URL and the encoding properties.

**Comment by XeR** [comment]
Hi, sorry for the delay. My previous work contract made it very hard for me to contribute code to open-source projects.
This is solved by 5507d16f430c091f273b7b29ed4e7c55eef54998
